src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        Map<Class<? extends Machine>, List<Machine>> mapMachines = new HashMap<>();
+        mapMachines.put(Bulldozer.class, new BulldozerProducer().get());
+        mapMachines.put(Excavator.class, new ExcavatorProducer().get());
+        mapMachines.put(Truck.class, new TruckProducer().get());
```

Comment:
you are creating this map each time the method is called. Let's make 'machines' a field and initialize it directly there using Map.of() or inside a constructor